### DISCUSSION TITLE
Add option to automatically start Rustdesk on startup even if app was closed.

### DISCUSSION THREAD

**Discussion by breisig** [rustdesk/rustdesk#9108] [General]
One feature that would be really good is having and option to automatically have rustdesk run at startup. It seems with the current version [1.2.7 or daily version 1.3.0] under wWndows 11 pro [23h2], if I run the app and leave it running on the taskbar and reboot it starts up fine on reboot. However, if you right click it on the task bar and close the app [Example: Just to close it temporarily] and reboot, Rustdesk will not automatically run at startup. Other apps like teamviewer or vnc have an option that will allow it to run automatically on startup even if you close the app. This is useful especially if someone accidentally closes apps and not realize that it doesn't run on startup. Hopefully this should be an easier request to add.

Installing rustdesk and it installs it and runs as a service which is fine and if I reboot the computer, rustdesk will startup and run on the taskbar meaning that it's running after a reboot. That part is fine. The REAL issue is if someone wants to 'temporary' stop it by right click the rustdesk icon on the taskbar and selects exit [which stops it from running temporarily] and then reboot your computer, rustdesk will *not* run again on startup as you won't see it on the taskbar anymore. This is a HUGE issue as I know from time to time, people want to temporary stop rustdesk from running but no matter what, we *always* want it to automatically start at startup.

**Comment by liamf91** [comment]
When installing I believe you have the option to enable "run on startup" I have personally never had this issue on Windows 11 Pro or even MacOS.

**Reply by jayjohnson1976** [reply]
When closing Rustdesk from the lower system tray, you have to start rustdesk
 and manually restart the service on the bottom of the GUI. Then it will run like it did before.
![Russtdesk](https://github.com/user-attachments/assets/8d627930-e76d-4615-bee4-f703ca8283a5)

**Comment by pdifeo** [comment]
> ... under wWndows 11 pro [23h2], if I run the app and leave it running on the taskbar and reboot it starts up fine on reboot. 

This is because you have fast startup enabled.

**Reply by breisig** [reply]
The REAL issue is if someone wants to 'temporary' stop it by right click the rustdesk icon on the desktop and selects exit [which stops it from running temporarily] and then reboot your computer, rustdesk will *not* run again on startup as you won't see it on the taskbar anymore after.

I don't see any 'fast startup' options in the settings area.

**Reply by pdifeo** [reply]
fast startup is a Windows feature. It not shutdown all applications when poweroff the computer but hibernate them. So when the computer restart, you have all application running as before.

**Reply by jvschiavo** [reply]
> fast startup is a Windows feature. It not shutdown all applications when poweroff the computer but hibernate them. So when the computer restart, you have all application running as before.

This is only the case if you poweroff/shutdown and then turn on the computer again. This *isn't* the case when you just click to reboot.

**Comment by breisig** [comment]
Installing rustdesk and it installs it and runs as a service which is fine and if I reboot the computer, rustdesk will startup and run on the taskbar meaning that it's running after a reboot. That part is fine. The REAL issue is if someone wants to 'temporary' stop it by right click the rustdesk icon on the taskbar and selects exit [which stops it from running temporarily] and then reboot your computer, rustdesk will *not* run again on startup as you won't see it on the taskbar anymore. This is a HUGE issue as I know from time to time, people want to temporary stop rustdesk from running during their session but no matter what, we *always* want it to automatically run at startup

**Comment by Hugo1380** [comment]
Have you planned to implement this feature? Or is there a workaround?

My use case is to have the rustdesk client running on all my devices to easily connect to them with a persistent password. So I need to ensure that when a device is rebooted, the rustdesk client + service is automatically launched.

**Reply by jvschiavo** [reply]
If you don't close the app this will work.

But this is different from what the OP is asking, though.

**Comment by muhammadomais** [comment]
Our particular situation is that we are putting rust desk at different location those pc's don't have the monitor attached. So we go though teamviewer and open Rust Desk to use it. We are trying this to move away from Teamviewer but we are still dependent on that. If their is a solution to that problem please let me know. Any help is highly appreciated.

**Comment by dlukt** [comment]
How DO I set it to always start?
I thought it's a service, why isn't it listed in windows services?
I'm relying on this to work 100% reliably.
I don't have fast boot enabled.

**Comment by jayjohnson1976** [comment]
Not sure bud, all I can say is one time I closed it by right clicking the
icon down in the bottom right system tray. Then off-site noticed it wasn't
running to log in remotely. After some research noticed a post from another
person and saw it's right on the bottom of the main interface. Doesn't run
again until you click it to start the service again.

Simple and so easy, had same issue and this is how I restarted my service.
Not sure why but it's right in front of everyone, not sure a lot of people
know about that. The other way if restarting it was to uninstall it and
reinstall it again. I said to myself gotta be a better way. This is all the
info I have for you on the matter. Easy fix.

Ttyl

Jason

**Reply by dlukt** [reply]
Yeah the solution is not obvious.
If you want the service to be installed and running on every boot, don't close Rustdesk but quitting via the tray icon exit/quit.
Regular [x]ing does not delete the service, but exiting via tray icon does.
This is unusual.
The default behavior should be that the service is always running unless explicitly uninstalled.

**Comment by jayjohnson1976** [comment]
Yes I agree with what your saying, it is unusual behaviour for closing an
app via the system tray. I was baffled when I was offsite trying to remote
in. After digging in found a post about it being on the GUI.

It should be changed, starts upon reboot. Only truly disable if you stop or
disable the service in the task manager.

Jason

**Comment by etegration** [comment]
Sep 2025 and I am still googling for this. Obviously it’s a design problem. Can we have auto start no matter what? Unless we uninstall.

**Comment by jvschiavo** [comment]
Do we have an issue to follow about this feature besides this discussion?

**Reply by etegration** [reply]
You and i would think it’s an easy fix…. No news yet.

**Reply by jvschiavo** [reply]
> You and i would think it’s an easy fix…. No news yet.

I just mean an issue thread would be better to follow than a "discussion". This discussion feature of GitHub is terrible for a lot of things that should keep being an issue thread imho.

**Comment by etegration** [comment]
Shit. Happen to me again. Forgot to start it after reboot? How hard is it to have an option to auto start Rustdesk? Damn it. It’s not about finding a way to write our own script. Add start up. Do auto detection scripts whatever. It should be built in fk.

**Comment by southerndoc** [comment]
I would like to see this implemented with macOS.  It restarts on reboot, but if you've closed the app prior to restart, it won't load again.  I would love for it to load by default.
